moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -354,32 +355,33 @@ bool RuckigSmoothing::runRuckig(robot_trajectory::RobotTrajectory& trajectory,
         duration_extension_factor *= DURATION_EXTENSION_FRACTION;
         // Reset the trajectory
         trajectory = robot_trajectory::RobotTrajectory(original_trajectory, true /* deep copy */);
-        for (size_t time_stretch_idx = 1; time_stretch_idx < num_waypoints; ++time_stretch_idx)
+
+        size_t time_stretch_idx = waypoint_idx;
+        trajectory.setWayPointDurationFromPrevious(
+            time_stretch_idx + 1,
+            duration_extension_factor * original_trajectory.getWayPointDurationFromPrevious(time_stretch_idx + 1));
+        // re-calculate waypoint velocity and acceleration
+        auto target_state = trajectory.getWayPointPtr(time_stretch_idx + 1);
+        const auto prev_state = trajectory.getWayPointPtr(time_stretch_idx);
+        timestep = trajectory.getAverageSegmentDuration();
+        for (size_t joint = 0; joint < num_dof; ++joint)
         {
-          trajectory.setWayPointDurationFromPrevious(
-              time_stretch_idx,
-              duration_extension_factor * original_trajectory.getWayPointDurationFromPrevious(time_stretch_idx));
-          // re-calculate waypoint velocity and acceleration
-          auto target_state = trajectory.getWayPointPtr(time_stretch_idx);
-          const auto prev_state = trajectory.getWayPointPtr(time_stretch_idx - 1);
-          timestep = trajectory.getAverageSegmentDuration();
-          for (size_t joint = 0; joint < num_dof; ++joint)
-          {
-            target_state->setVariableVelocity(move_group_idx.at(joint),
-                                              (1 / duration_extension_factor) *
-                                                  target_state->getVariableVelocity(move_group_idx.at(joint)));
-
-            double prev_velocity = prev_state->getVariableVelocity(move_group_idx.at(joint));
-            double curr_velocity = target_state->getVariableVelocity(move_group_idx.at(joint));
-            target_state->setVariableAcceleration(move_group_idx.at(joint), (curr_velocity - prev_velocity) / timestep);
-          }
-          target_state->update();
+          target_state->setVariableVelocity(move_group_idx.at(joint),
+                                            (1 / duration_extension_factor) *
+                                                target_state->getVariableVelocity(move_group_idx.at(joint)));
+
+          double prev_velocity = prev_state->getVariableVelocity(move_group_idx.at(joint));
+          double curr_velocity = target_state->getVariableVelocity(move_group_idx.at(joint));
+          target_state->setVariableAcceleration(move_group_idx.at(joint), (curr_velocity - prev_velocity) / timestep);
         }
+        target_state->update();
+
         ruckig_ptr = std::make_unique<ruckig::Ruckig<ruckig::DynamicDOFs>>(num_dof, timestep);
         initializeRuckigState(*trajectory.getFirstWayPointPtr(), group, ruckig_input, ruckig_output);
         // Begin the while() loop again
```

Comment:
Let's update this comment. It's confusing b/c there are 2 while loops now:

```suggestion
        // Try a larger duration_extension_factor
```